lib/shared/inference/openai_image_generator.rb

```diff
@@ -5,23 +5,31 @@ module Inference
     class OpenAiImageGenerator
       TIMEOUT = 60
 
-      def self.perform!(prompt, model: "dall-e-3", size: "1024x1024", api_key: nil)
+      def self.perform!(prompt, model: "dall-e-3", size: "1024x1024", api_key: nil, api_url: nil)
         api_key ||= SiteSetting.ai_openai_api_key
+        api_url ||= SiteSetting.ai_openai_dall_e_3_url
 
-        url = URI("https://api.openai.com/v1/images/generations")
-        headers = { "Content-Type" => "application/json", "Authorization" => "Bearer #{api_key}" }
+        uri = URI(api_url)
+
+        headers = { "Content-Type" => "application/json", "quality" => "hd" }
```

Comment:
Interesting choice to have the `quality` as a header, or is supposed to be part of the body?

Reply:
oh ooops good catch will fix